### PR TITLE
Tickets/dm 39289

### PR DESCRIPTION
**Pull request by iagaponenko**

### CODE REVIEW

File: src/replica/HttpDirectorIndexModule.cc [hunk 1]
```diff
@@ -78,123 +78,114 @@ json HttpDirectorIndexModule::_buildDirectorIndex() {
...
-            queries.push_back(g.dropTable(indexTableName, ifExists));
+
+    // The object will stay empty if no problems will be encountered during the construction
+    // of the index.
+    json extError = json::object();
+
+    // Manage the new connection via the RAII-style handler to ensure the transaction
+    // is automatically rolled-back in case of exceptions.
+    ConnectionHandler const h(qservMasterDbConnection("qservMeta"));
+    QueryGenerator const g(h.conn);
```
Comment (jgates108) [inline src/replica/HttpDirectorIndexModule.cc:143]: I'm going to argue against the single letter variable name. There's no way to search for 'h' or 'g` reliably. 'conH' and 'qGen' would be significant improvements.

**Review by jgates108 — APPROVED** [review]
Looks good, just some minor coments.

File: src/replica/HttpDirectorIndexModule.cc (hunk 1 above)
Comment (iagaponenko) [reply src/replica/HttpDirectorIndexModule.cc:143]: Good point. Will fix it.

File: src/replica/HttpDirectorIndexModule.cc [hunk 2]
```diff
@@ -78,123 +78,114 @@ json HttpDirectorIndexModule::_buildDirectorIndex() {
...
-        if (table.directorTable.primaryKeyColumn().empty()) {
-            string const msg =
-                    "director table has not been properly configured in database '" + database.name + "'";
-            throw HttpError(__func__, msg);
-        }
-        primaryKeyColumn[table.name] = table.directorTable.primaryKeyColumn();
+    auto const table = database.findTable(directorTableName);
+    if (!table.isDirector) {
+        string const msg = "table '" + table.name +
+                           "' is not configured as the director table in database '" + database.name + "'";
```
Comment (fritzm) [inline src/replica/HttpDirectorIndexModule.cc:89]: Should be "... not configured as a director table ..." (there can be more than one director in the database)

**Review by fritzm — APPROVED** [review]
LGTM -- one very minor comment change recommended

File: src/replica/HttpDirectorIndexModule.cc (hunk 2 above)
Comment (iagaponenko) [reply src/replica/HttpDirectorIndexModule.cc:89]: Good point!

**Pull request merged**